src/main/java/mate/academy/EventManager.java

```diff
@@ -1,19 +1,29 @@
 package mate.academy;
 
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
 public class EventManager {
-    public void registerListener(EventListener listener) {
+    private final List<EventListener> listeners = new CopyOnWriteArrayList<>();
+    private final ExecutorService executorService = Executors.newCachedThreadPool();
```

Comment:
Using Executors.newCachedThreadPool() can potentially create a large number of threads if many events are being fired in quick succession. This could lead to performance issues. Consider using a fixed thread pool or other types of executors that limit the number of threads.

Reply:
Not resolved